### PR TITLE
move version to vs2019, the 1903 sdk, and the 14.2 build tools.

### LINKED ISSUE #565: Move the solution to VS2019 and v142 toolset
As mentioned in README, (also related to #534) 

> If you're running VS2019, you'll also need to install the "v141 Toolset" and "Visual C++ ATL for x86 and x64"

But actually v141 Toolset is not required in VS2019. The following are the configurations of my VS2019 and I can build the solution successfully. 

C++ Desktop development
image

UWP development
image

Individual components
image

To build the solution in VS2019, you need to have:
- Windows 10 version 1903
- Windows 10 SDK (10.0.18362.0)
- VS2019 with C++ Desktop and UWP development workloads, as shown in the above screenshots

And **retarget the solution** to v142 platform and the latest Windows SDK.

### PR DESCRIPTION
**Pull request by miniksa**
## Summary of the Pull Request
Moves the solution to VS2019 and the latest SDK/tools

## PR Checklist
* [x] Closes #565
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Tests added/passed
* [x] Does not require documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #565 

## Detailed Description of the Pull Request / Additional comments
This is sitting here until we figure out what to do with the build agents regarding versioning and signing.

### CODE REVIEW

**Comment by miniksa** [comment]
WELP. Looks like the 2019 hosted pool doesn't have the 1903 SDK yet either...

https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/Vs2019-Server2019-Readme.md

**Comment by miniksa** [comment]
Dependent on https://github.com/microsoft/azure-pipelines-image-generation/issues/734 and the discussion in https://github.com/microsoft/azure-pipelines-image-generation/pull/770

**Comment by Biswa96** [comment]
In VS2019, we don't have to specify SDK full version number. VS selects latest version. For example, we can use 10 instead of 10.0.18362.0.

**Comment by InsightfulEthics** [comment]
I ran this locally on my machine after I saw that it failed the pipeline as the pipeline doesn’t have the v142 tool kit and the build was a success! I downloaded the zip from this Archive and set it up just as mentioned in #489 as #565 only worked the first build then failed afterwards. In this attempt I specifically retargeted the build to the 10.0.18362.0 SDK and checked the toolkit to make sure it was targeted to v142 despite knowing it is now predefined in the code. I let it run and it came back with a few warnings that are all common in the other pipeline builds but not of massive concern as the build did succeed. I certainly do hope to see this solution work for others. Keep up the great work!

**Comment by miniksa** [comment]
18362 is coming...

https://github.com/microsoft/azure-pipelines-image-generation/pull/953

**Comment by oising** [comment]
If there ever was a pipeline that's in dire need of azure devops build cache save/restore tasks, this one is it. Apparently these things are real, and coming pretty soon.

File: src/cascadia/CascadiaPackage/CascadiaPackage.wapproj [hunk 1]
```diff
@@ -142,29 +131,20 @@
...
   <!-- Stomp the path to conhost.exe.
     This task will copy OpenConsole.exe to the appx as conhost.exe, and place it
     adjacent to WindowsTerminal.exe.
    -->
   <Target Name="OpenConsoleStompSourceProjectForWapProject" BeforeTargets="_ConvertItems">
-    <ItemGroup>
-      <!-- Stomp all "SourceProject" values for all incoming dependencies to flatten the package. -->
-      <_TemporaryFilteredWapProjOutput
-        Include="@(_FilteredNonWapProjProjectOutput)" />
+    <ItemGroup><!-- Stomp all "SourceProject" values for all incoming dependencies to flatten the package. -->
```
Comment (DHowett-MSFT) [inline src/cascadia/CascadiaPackage/CascadiaPackage.wapproj]: i guess while you're changing this file, these are actually spaces that are encoded wrong (!) and all need to be fixed

**Comment by miniksa** [comment]
> If there ever was a pipeline that's in dire need of azure devops build cache save/restore tasks, this one is it. Apparently these things are real, and coming pretty soon.

Say what now? Can you provide a link to this?

File: src/cascadia/CascadiaPackage/CascadiaPackage.wapproj (hunk 1 above)
Comment (miniksa) [reply src/cascadia/CascadiaPackage/CascadiaPackage.wapproj]: I think I got it.

**Comment by andersosthus** [comment]
@miniksa Looks like it's available for testing for internal orgs: https://github.com/Microsoft/azure-pipelines-yaml/pull/113

**Comment by miniksa** [comment]
@andersosuthus, thanks. We're not operating as an internal org right now, so I'll probably wait on this like the rest of everyone in the public.

**Comment by DHowett-MSFT** [comment]
I'm going to tee this up again now that all our dependencies have merged.

**Comment by illera88** [comment]
any update on this?

**Comment by DHowett-MSFT** [comment]
We're waiting on the 18362 SDK to be deployed to the Azure DevOps tenant from which we're running our CI builds.

File: build/pipelines/templates/build-console-ci.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
   variables:
     BuildConfiguration: ${{ parameters.configuration }}
     BuildPlatform: ${{ parameters.platform }}
-  pool: { vmImage: vs2017-win2016 }
+  pool: { vmImage: windows-2019 }
```
Comment (adiviness) [inline build/pipelines/templates/build-console-ci.yml]: this change should be also done to `build-console-audit-job.yml` and `check-formatting.yml`

**Pull request merged**